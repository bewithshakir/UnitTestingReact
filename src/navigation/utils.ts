
 export const combinePaths = (parent: string, child: string): string =>
 `${parent.replace(/\/$/, "")}/${child.replace(/^\//, "")}`;


export const buildPaths = (routes:any, parentPath: string = ""): any =>
 routes.map((route: { path: string; routes: any; }) => {
   const path = combinePaths(parentPath, route.path);

   return {
     ...route,
     path,
     ...(route.routes && { routes: buildPaths(route.routes, path) })
   };
 });


export const setupParents = (routes:any, parentRoute:any = null): any =>
 routes.map((route: any) => {
   const withParent = {
     ...route,
     ...(parentRoute && { parent: parentRoute })
   };

   return {
     ...withParent,
     ...(withParent.routes && {
       routes: setupParents(withParent.routes, withParent)
     })
   };
 });

export const flattenRoutes = (routes:[]):any =>
 routes
   .map((route: { routes: any; }) => [route.routes ? flattenRoutes(route.routes) : [], route])
   .flat(Infinity);

/**
* Combine all the above functions together
*
* @param routes
* @returns {any[]}
*/
export const generateAppRoutes = (routes: any) => {
 return flattenRoutes(setupParents(buildPaths(routes)));
};

/**
* Provides path from root to the element
*
* @param route
* @returns {any[]}
*/
export const pathTo = (route: { parent: any; }) : any=> {
 if (!route.parent) {
   return [route];
 }

 return [...pathTo(route.parent), route];
};