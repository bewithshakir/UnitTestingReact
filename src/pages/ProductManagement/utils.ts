const setHelperTextByFieldType = (formik: any, name: string, fieldType: string) => {
    if (fieldType === 'select') {
        return (formik.errors as any)[name]?.value;
    }
    if (fieldType === 'input') {
        return (formik.errors as any)[name];
    }
    if (fieldType === 'multiselect') {
        return JSON.parse(JSON.stringify(formik.errors[name])) ;
    }
    return undefined;
};

export const getFormFieldHelperText = (formik: any, name: string, fieldType: string) => {
    if ((formik.touched as any)[name] && (formik.errors as any)[name]) {
        return setHelperTextByFieldType(formik, name, fieldType);
    }
    return undefined;
};

export const getFormFieldError = (formik: any, name: string) => {
    if ((formik.touched as any)[name] && (formik.errors as any)[name]) {
        return true;
    }
    return false;
};