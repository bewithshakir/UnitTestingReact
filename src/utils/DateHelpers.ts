import moment from 'moment';

const preFormatDate = (date: string | null | Date) => {
  return typeof date === 'string' && date.length > 0 ? date.replace('Z', '') : date;
};

export const formatDateAsYYYYMMDD = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('YYYY-MM-DD') : '';
};

export const formatTime = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('hh:mm A') : '';
};

export const formatDateAsMMDDYYYY = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('MM/DD/YYYY') : '';
};

export const formatDateMonthDayYear = (date: string | null | Date) => {
  return date ? moment(preFormatDate(date)).format('MMM DD, YYYY') : '';
};

export const formatDateTime = (date: string | null | Date) => {
  return date ? moment(date).format('MMM DD, YYYY hh:mm A') : '';
};

export const isTheDateToday = (date: string | null | Date) => {
  return date ? moment(date).format("Do") === moment().format("Do") : '';
};

export const getCurrentYear = () => {
  return new Date().getFullYear();
};