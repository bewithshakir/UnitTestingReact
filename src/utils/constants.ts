export const maxContacts = 5;
export const pageDataLimit = 15;

export const ParkingLot_SearchParam = 'parkinglotlanding';