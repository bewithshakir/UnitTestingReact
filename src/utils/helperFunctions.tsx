import { useTheme } from '../contexts/Theme/Theme.context';
import {
    YellowFuelIcon, RedFuelIcon, GreenFuelIcon, NavyBlueFuelIcon, ParrotGreenFuelIcon,
    PurpleFuelIcon, SkyBlueFuelIcon, BrownFuelIcon, OrangeFuelIcon, AquaFuelIcon
} from '../assets/icons';

type fileFormatSize = 'Bytes'|'KB'|'MB';
export const maskPhoneNumber = (phNumber: string) => {
    const maskedPh = phNumber.match(/(\d{3})(\d{3})(\d{4})/);
    return maskedPh ? "(" + maskedPh[1] + ") " + maskedPh[2] + "-" + maskedPh[3] : phNumber;
};

export const formatFileSizeUnit = (sizeInBytes: number, fileSizeUnit: fileFormatSize = 'MB') => {
    let coversionParam = null ; 
    switch(fileSizeUnit) {
        case 'Bytes' :
            coversionParam = 1;
          break;
        case 'KB':
            coversionParam = 1000;
          break;
          case 'MB':
            coversionParam = 1000*1000;
          break;
        default:
            coversionParam = 1;
      }
      return parseFloat((sizeInBytes/(coversionParam)).toFixed(2)) + ` ${fileSizeUnit}`;
};

export const getCheckBoxDisabledByPaymentType = (value: string) => {
    return {
        'WEX': false,
        'Invoice': true,
        'Internal': true,
        'Voyager': true,
        'default': true,
    }[value || 'default'];
};

export const getSeachedDataTotalCount = (data: { pages: [{ data: { pagination: { totalCount: number } } }] }, msg: string[]) => {
    const totalCount = data?.pages[0]?.data?.pagination?.totalCount || 0;
    return (`${totalCount} ${totalCount > 1 ? msg[1] : msg[0]}`);
};


export const getCurrentAppCountry = () => {
    const { themeType } = useTheme();
    return themeType;
};

export const isCurrentAppCountryUSA = () => {
    return getCurrentAppCountry() === "USA";
};


export function getProductIcon(fuelStatus: string) {
    return ({
        "Yellow": YellowFuelIcon,
        "Red": RedFuelIcon,
        "Green": GreenFuelIcon,
        "Navy Blue": NavyBlueFuelIcon,
        "Parrot Green": ParrotGreenFuelIcon,
        "Sky Blue": SkyBlueFuelIcon,
        "Purple": PurpleFuelIcon,
        "Brown": BrownFuelIcon,
        "Orange": OrangeFuelIcon,
        "Aqua": AquaFuelIcon,
    }[fuelStatus] || YellowFuelIcon);
}

export function getLegendFontColor(fuelStatus: string) {
    const { theme } = useTheme();
    return ({
        "Yellow": theme["--Darkgray"],
        "Red": theme["--White"],
        "Green": theme["--White"],
        "Navy Blue": theme["--White"],
        "Parrot Green": theme["--Darkgray"],
        "Sky Blue": theme["--White"],
        "Purple": theme["--White"],
        "Brown": theme["--White"],
        "Orange": theme["--Darkgray"],
        "Aqua": theme["--Darkgray"],
    }[fuelStatus]);
}

export function getUploadedBy() {
    return 'Dinesh Chakkravarthi';
}

export function getUploadedIn(url: string) {
    if (url.includes('addCustomer') || url.includes('viewCustomer')) {
        return 'Add Customer';
    }
    return 'Attachments';
}

export function capitalizeFirstLetter(str: string) {
    return str && str.charAt(0).toUpperCase() + str.substring(1);
}