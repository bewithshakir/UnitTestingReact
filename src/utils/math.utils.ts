
export function truncateDecimals(num: number, decimalPlace: number) {
  const numS = num.toString();
  const decPos = numS.indexOf('.');
  const substrLength = decPos == -1 ? numS.length : 1 + decPos + decimalPlace;
  const trimmedResult = Number(numS.substr(0, substrLength));
  const finalResult = isNaN(trimmedResult) ? 0 : trimmedResult;
  return finalResult;
}
