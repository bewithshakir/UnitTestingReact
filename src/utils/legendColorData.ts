import { filter } from 'lodash';

import { getProductIcon } from "./helperFunctions";

export const formatLegendsData = (data: any)=> {
    const filteredData = filter(data?.pages[0].data.products, {  activeInactiveInd: 'Y' });

    const updatedData = filteredData.map(item => ({
        value: item.productCd,
        label: item.ProductGroup.productGroupNm + ' - ' + item.productNm, 
        icon: getProductIcon(item.ProductIcon.productIconNm),
        isDisabled: true
    }));
    return updatedData;
};

