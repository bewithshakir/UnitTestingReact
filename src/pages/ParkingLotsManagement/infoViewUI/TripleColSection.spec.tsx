import { renderWithClient } from '../../../tests/utils';
import TripleColSection from './TripleColSection.component';

const tripleSectionCompData = {
  "heading": "Pricing",
  "type": "3-col",
  "data": {
    "Diesel": {
      "key": "Diesel",
      "subKey": "(Custom name)",
      "midValue": "16.8416",
      "endValue": "OPIS Rack"
    },
    "Regular1 Retail": {
      "key": "Regular1 Retail",
      "subKey": "(Custom name)",
      "midValue": "5.4540",
      "endValue": "OPIS Retail"
    }
  }
};


describe('load triple col section component', () => {
  it('load component', async () => {
    const result = renderWithClient(<TripleColSection data={tripleSectionCompData} />);
    expect(result.container.querySelector('.sectionBox')).toBeInTheDocument();
  });
});
