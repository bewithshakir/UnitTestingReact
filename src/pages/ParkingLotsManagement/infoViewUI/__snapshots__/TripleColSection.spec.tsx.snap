// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load triple col section component load component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sectionBox"
      >
        <h4>
          Pricing
           
        </h4>
        <div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 css-sa6qj6-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowLabel tripleColSectionRowLabel css-gj1fbr-MuiGrid-root"
            >
              Diesel
              <div
                class="subKey"
              >
                (Custom name)
                 
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue sectionRowMidValue css-gj1fbr-MuiGrid-root"
            >
              $16.8416/gal
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue tripleColSectionRowValue css-gj1fbr-MuiGrid-root"
            >
              OPIS Rack
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth sectionRowDivider css-9mgopn-MuiDivider-root"
          />
        </div>
        <div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 css-sa6qj6-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowLabel tripleColSectionRowLabel css-gj1fbr-MuiGrid-root"
            >
              Regular1 Retail
              <div
                class="subKey"
              >
                (Custom name)
                 
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue sectionRowMidValue css-gj1fbr-MuiGrid-root"
            >
              $5.4540/gal
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue tripleColSectionRowValue css-gj1fbr-MuiGrid-root"
            >
              OPIS Retail
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth sectionRowDivider css-9mgopn-MuiDivider-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sectionBox"
    >
      <h4>
        Pricing
         
      </h4>
      <div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 css-sa6qj6-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowLabel tripleColSectionRowLabel css-gj1fbr-MuiGrid-root"
          >
            Diesel
            <div
              class="subKey"
            >
              (Custom name)
               
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue sectionRowMidValue css-gj1fbr-MuiGrid-root"
          >
            $16.8416/gal
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue tripleColSectionRowValue css-gj1fbr-MuiGrid-root"
          >
            OPIS Rack
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth sectionRowDivider css-9mgopn-MuiDivider-root"
        />
      </div>
      <div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-grid-xs-12 css-sa6qj6-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowLabel tripleColSectionRowLabel css-gj1fbr-MuiGrid-root"
          >
            Regular1 Retail
            <div
              class="subKey"
            >
              (Custom name)
               
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue sectionRowMidValue css-gj1fbr-MuiGrid-root"
          >
            $5.4540/gal
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 sectionRowValue tripleColSectionRowValue css-gj1fbr-MuiGrid-root"
          >
            OPIS Retail
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth sectionRowDivider css-9mgopn-MuiDivider-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
